client/signup/steps/site-picker/index.jsx

```diff
@@ -0,0 +1,87 @@
+/**
+ * External dependencies
+ */
+import React, { Component } from 'react';
+
+/**
+ * Internal dependencies
+ */
+import Card from 'components/card';
+import SiteSelector from 'components/site-selector';
+import StepWrapper from 'signup/step-wrapper';
+import sitesFactory from 'lib/sites-list';
+import SignupActions from 'lib/signup/actions';
+
+const sites = sitesFactory();
+
+class SitePicker extends Component {
+	constructor( props ) {
+		super( props );
+
+		this.handleSiteSelect = this.handleSiteSelect.bind( this );
+	}
+
+	handleSiteSelect( siteSlug ) {
+		const {
+				stepSectionName,
+				stepName,
+				goToStep,
+			} = this.props,
+			site = sites.getSite( siteSlug ),
+			hasPlan = site && site.plan && site.plan.product_slug !== 'free_plan';
+
+		SignupActions.submitSignupStep(
+			{
+				stepName,
+				stepSectionName,
+				siteId: site.ID,
+				siteSlug: site.slug
+			},
+			[],
+			{}
+		);
+
+		SignupActions.submitSignupStep( { stepName: 'themes', wasSkipped: true }, [], {
+			themeSlugWithRepo: 'pub/twentysixteen'
+		} );
+
+		if ( hasPlan ) {
+			SignupActions.submitSignupStep( { stepName: 'plans', wasSkipped: true }, [], { cartItem: null, privacyItem: null } );
+
+			goToStep( 'user' );
+		} else {
+			goToStep( 'plans' );
+		}
+	}
+
+	renderScreen() {
+		return (
+			<Card className="site-picker__wrapper">
+				<SiteSelector
+					filter={
+						function( site ) {
```

Comment:
While I was shamelessly copying this part of your PR for my own, I noticed that this filter should be a function defined in this class, not one inline here - otherwise on every render, a new function will be created, throwing off React and possibly causing unnecessary re-renders.